deepdoc/parser/monkeyocrv2_parser.py

```
@@ -0,0 +1,72 @@
+
+    def _convert_zip(self, archive_bytes):
+        sections, tables = [], []
+        with zipfile.ZipFile(io.BytesIO(archive_bytes)) as archive:
+            names = archive.namelist()
+            roots = {
+                name.split("/", 1)[0]
+                for name in names
+                if "/" in name
+                and (
+                    name.endswith(".md")
+                    or "/jsons/" in name and name.endswith(".json")
```

Ruff (0.16.2)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)